actor/v2action/service_access.go

```diff
@@ -0,0 +1,223 @@
+package v2action
+
+import (
+	"fmt"
+
+	"code.cloudfoundry.org/cli/actor/actionerror"
+	"code.cloudfoundry.org/cli/api/cloudcontroller/ccv2"
+	"code.cloudfoundry.org/cli/api/cloudcontroller/ccv2/constant"
+)
+
+// EnableServiceForAllOrgs enables access for the given service in all orgs.
+func (actor Actor) EnableServiceForAllOrgs(serviceName string) (Warnings, error) {
+	var allWarnings Warnings
+
+	services, warnings, err := actor.CloudControllerClient.GetServices(ccv2.Filter{
+		Type:     constant.LabelFilter,
+		Operator: constant.EqualOperator,
+		Values:   []string{serviceName},
+	})
+
+	allWarnings = append(allWarnings, warnings...)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	if len(services) == 0 {
+		return allWarnings, actionerror.ServiceNotFoundError{Name: serviceName}
+	}
+
+	servicePlans, warnings, err := actor.CloudControllerClient.GetServicePlans(ccv2.Filter{
+		Type:     constant.ServiceGUIDFilter,
+		Operator: constant.EqualOperator,
+		Values:   []string{services[0].GUID},
+	})
+
+	allWarnings = append(allWarnings, warnings...)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	for _, plan := range servicePlans {
+		warnings, err = actor.CloudControllerClient.UpdateServicePlan(plan.GUID, true)
+		allWarnings = append(allWarnings, warnings...)
+		if err != nil {
+			return allWarnings, err
+		}
+	}
+
+	return allWarnings, nil
+}
+
+// EnablePlanForAllOrgs enables access to a specific plan of the given service in all orgs.
+func (actor Actor) EnablePlanForAllOrgs(serviceName, servicePlan string) (Warnings, error) {
+	var allWarnings Warnings
+
+	services, warnings, err := actor.CloudControllerClient.GetServices(ccv2.Filter{
+		Type:     constant.LabelFilter,
+		Operator: constant.EqualOperator,
+		Values:   []string{serviceName},
+	})
+	allWarnings = append(allWarnings, warnings...)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	if len(services) == 0 {
+		return allWarnings, actionerror.ServiceNotFoundError{Name: serviceName}
+	}
+
+	servicePlans, warnings, err := actor.CloudControllerClient.GetServicePlans(ccv2.Filter{
+		Type:     constant.ServiceGUIDFilter,
+		Operator: constant.EqualOperator,
+		Values:   []string{services[0].GUID},
+	})
+	allWarnings = append(allWarnings, warnings...)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	for _, plan := range servicePlans {
+		if plan.Name == servicePlan {
+			warnings, err := actor.removeOrgLevelServicePlanVisibilities(plan.GUID)
```

Comment:
It would be great if you add the comments from the legacy code. It was difficult for us to understand why we need to delete plan visibilities without this comment.
https://github.com/cloudfoundry/cli/blob/4445fd548dea41993dc4b9f474830f1ce7aa2d91/cf/actors/services_plans.go#L180-L181

Reply:
Fine with me.
Even with the comment I don't really get why we do it.
Looks to me more like a CC problem, not CLI. However we kept the behaviour in order to be consistent with the old command.

Reply:
"Looks to me more like a CC problem, not CLI. However we kept the behaviour in order to be consistent with the old command." <-- I agree.